cmd/pomerium/main.go

```diff
@@ -199,58 +144,61 @@ func wrapMiddleware(o *config.Options, mux http.Handler) http.Handler {
 	return c.Then(mux)
 }
 
-func parseOptions(configFile string) (*config.Options, error) {
-	o, err := config.OptionsFromViper(configFile)
-	if err != nil {
-		return nil, err
-	}
-	if o.Debug {
-		log.SetDebugMode()
-	}
-	if o.LogLevel != "" {
-		log.SetLevel(o.LogLevel)
-	}
-	metrics.AddPolicyCountCallback(o.Services, func() int64 {
-		return int64(len(o.Policies))
-	})
-	checksumInt, err := strconv.ParseInt(fmt.Sprintf("0x%s", o.Checksum()), 0, 64)
-	if err != nil {
-		log.Warn().Err(err).Msg("Could not parse config checksum into integer")
+func configToServerOptions(opt *config.Options) *httputil.ServerOptions {
+	return &httputil.ServerOptions{
+		Addr:              opt.Addr,
+		Cert:              opt.Cert,
+		Key:               opt.Key,
+		CertFile:          opt.CertFile,
+		KeyFile:           opt.KeyFile,
+		ReadTimeout:       opt.ReadTimeout,
+		WriteTimeout:      opt.WriteTimeout,
+		ReadHeaderTimeout: opt.ReadHeaderTimeout,
+		IdleTimeout:       opt.IdleTimeout,
 	}
-	metrics.SetConfigChecksum(o.Services, checksumInt)
-	return o, nil
 }
 
-func handleConfigUpdate(opt *config.Options, services []config.OptionsUpdater) *config.Options {
-	newOpt, err := parseOptions(*configFile)
-	if err != nil {
-		log.Error().Err(err).Msg("cmd/pomerium: could not reload configuration")
-		metrics.SetConfigInfo(opt.Services, false, "")
-		return opt
+func setupMetrics(opt *config.Options) {
+	if opt.MetricsAddr != "" {
+		if handler, err := telemetry.PrometheusHandler(); err != nil {
+			log.Error().Err(err).Msg("cmd/pomerium: couldn't start metrics server")
+		} else {
+			serverOpts := &httputil.ServerOptions{Addr: opt.MetricsAddr}
+			srv := httputil.NewHTTPServer(serverOpts, handler)
```

Comment:
I like this pattern

Reply:
Awesome! Me too. I think this paves the way if/when we have to serve gRPC distinctly from our HTTP(s) handlers.